File: java/src/main/java/com/tormentnexus/claudecode/SlashCommandRouter.java
package com.tormentnexus.claudecode;

// Re-implementation of Claude Code's Slash Command Router

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.CompletableFuture;
import java.util.function.Function;

public class SlashCommandRouter {

    private final Map<String, Function<String[], CompletableFuture<String>>> commands = new HashMap<>();

    public void registerCommand(String command, Function<String[], CompletableFuture<String>> handler) {
        if (!command.startsWith("/")) {
            throw new IllegalArgumentException("Slash commands must start with '/'");
        }
        commands.put(command, handler);
    }

    public boolean isSlashCommand(String input) {
        return input.trim().startsWith("/");
    }

    public CompletableFuture<String> executeCommand(String input) {
        String[] parts = input.trim().split("\\s+");
        if (parts.length == 0 || parts[0].isEmpty()) {
            return CompletableFuture.completedFuture("Empty command.");
        }

        String cmd = parts[0];
        String[] args = Arrays.copyOfRange(parts, 1, parts.length);

        Function<String[], CompletableFuture<String>> handler = commands.get(cmd);
        if (handler != null) {
            return handler.apply(args);
        }

        return CompletableFuture.completedFuture("Unknown command: " + cmd + ". Type /help for available commands.");
    }
}

File: csharp/ClaudeCodeFeatures/SlashCommandRouter.cs
// Re-implementation of Claude Code's Slash Command Router

using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TormentNexus.ClaudeCodeFeatures
{
    public class SlashCommandRouter
    {
        private readonly Dictionary<string, Func<string[], Task<string>>> _commands = new Dictionary<string, Func<string[], Task<string>>>();

        public void RegisterCommand(string command, Func<string[], Task<string>> handler)
        {
            if (!command.StartsWith("/"))
            {
                throw new ArgumentException("Slash commands must start with '/'");
            }
            _commands[command] = handler;
        }

        public bool IsSlashCommand(string input)
        {
            return input.Trim().StartsWith("/");
        }

        public async Task<string> ExecuteCommand(string input)
        {
            var parts = input.Trim().Split(new[] { ' ', '\t' }, StringSplitOptions.RemoveEmptyEntries);
            if (parts.Length == 0) return "Empty command.";

            var cmd = parts[0];
            var args = parts.Skip(1).ToArray();

            if (_commands.TryGetValue(cmd, out var handler))
            {
                return await handler(args);
            }

            return $"Unknown command: {cmd}. Type /help for available commands.";
        }
    }
}
